src/main/kotlin/org/kamiblue/client/KamiMod.kt

```diff
@@ -0,0 +1,79 @@
+package org.kamiblue.client
+
+import net.minecraftforge.common.MinecraftForge
+import net.minecraftforge.fml.common.Mod
+import net.minecraftforge.fml.common.event.FMLInitializationEvent
+import net.minecraftforge.fml.common.event.FMLPostInitializationEvent
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent
+import org.apache.logging.log4j.LogManager
+import org.apache.logging.log4j.Logger
+import org.kamiblue.client.event.ForgeEventProcessor
+import org.kamiblue.client.gui.mc.KamiGuiUpdateNotification
+import org.kamiblue.client.util.ConfigUtils
+import org.kamiblue.client.util.threads.BackgroundScope
+import java.io.File
+
+@Mod(
+    modid = KamiMod.ID,
+    name = KamiMod.NAME,
+    version = KamiMod.VERSION
+)
+class KamiMod {
+
+    companion object {
```

Comment:
@l1ving @Xiaro shouldn't be just an object KamiMod instead of a class with a companion object?

Reply:
I have a feeling that forge wouldn't like it, but https://tryitands.ee

Reply:
object won't generate constructors. And forge needs to call the constructor to make an instance before assigning it to the INSTANCE field

Reply:
![image](https://user-images.githubusercontent.com/62033805/107715129-98409780-6c9c-11eb-802c-adb73e6e43ac.png)
Yes it needs the constructor so it can't be an `object`